packages/ckeditor5-utils/src/focustracker.ts

```diff
@@ -111,28 +243,160 @@ export default class FocusTracker extends /* #__PURE__ */ DomEmitterMixin( /* #_
 	 */
 	private _focus( element: Element ): void {
-		clearTimeout( this._nextEventLoopTimeout! );
+		// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": _focus() on element`, element );
+
+		this._clearBlurTimeout();
 
 		this.focusedElement = element;
 		this.isFocused = true;
 	}
 
 	/**
-	 * Clears currently focused element and set {@link #isFocused} as `false`.
-	 * This method uses `setTimeout` to change order of fires `blur` and `focus` events.
+	 * Clears currently {@link #focusedElement} and sets {@link #isFocused} `false`.
+	 * This method uses `setTimeout()` to change order of `blur` and `focus` events calls, ensuring that moving focus between
+	 * two elements within a single focus tracker's scope, will not cause `[ blurA, focusB ]` sequence but just `[ focusB ]`.
+	 * The former would cause a momentary change of `#isFocused` to `false` which is not desired because any logic listening to
+	 * a focus tracker state would experience UI flashes and glitches as the user focus travels across the UI.
 	 */
 	private _blur(): void {
-		clearTimeout( this._nextEventLoopTimeout! );
+		const isAnyElementFocused = this.elements.find( element => element.contains( document.activeElement ) );
+
+		// Avoid blurs originating from external FTs when the focus still remains in one of the #elements.
+		if ( isAnyElementFocused ) {
+			return;
+		}
+
+		const isAnyExternalViewFocused = this.externalViews.find( view => {
+			// Do not consider external views's focus trackers as focused if there's a blur timeout pending.
+			return view.focusTracker.isFocused && !view.focusTracker._blurTimeout;
+		} );
+
+		// Avoid unnecessary DOM blurs coming from #elements when the focus still remains in one of #externalViews.
+		if ( isAnyExternalViewFocused ) {
+			return;
+		}
+
+		this._clearBlurTimeout();
+
+		this._blurTimeout = setTimeout( () => {
+			// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": Blur.` );
 
-		this._nextEventLoopTimeout = setTimeout( () => {
 			this.focusedElement = null;
 			this.isFocused = false;
 		}, 0 );
 	}
+
+	/**
+	 * Clears the asynchronous blur event timeout on demand. See {@link #_blur} to learn more.
+	 */
+	private _clearBlurTimeout(): void {
+		clearTimeout( this._blurTimeout! );
+		this._blurTimeout = null;
+	}
+
+	// @if CK_DEBUG_FOCUSTRACKER // public static _instances: Array<FocusTracker> = [];
+}
+
+/**
+ * A {@link module:ui/view~View} instance with a {@link module:utils/focustracker~FocusTracker} instance exposed
+ * at the `#focusTracker` property.
+ */
+export type ViewWithFocusTracker = View & { focusTracker: FocusTracker };
+
+/**
+ * Checks whether a view is an instance of {@link ~ViewWithFocusTracker}.
+ */
+export function isViewWithFocusTracker( view: any ): view is ViewWithFocusTracker {
+	return 'focusTracker' in view && view.focusTracker instanceof FocusTracker;
+}
+
+function isElement( value: any ): value is Element {
```

Comment:
I didn't want to pollute this PR with unrelated changes, but it seems we have several similar functions scattered in the codebase. I'll create a follow-up and move this to `utils/src/dom` to share across the project.

Reply:
The reason may be that sometimes these functions were heavily used and `lodash` implementation of the method is very slow, it resulted in performance hiccups. And I know we use `lodash` to make sure that something is really DOM Element ---- AFAIR there were problems if the element came from `iframe` or something? I don't remember exactly.

But I remember that I've been changing some code to use a "simpler" check where I thought it is safe and where it was heavily used.